crates/forge/src/runner.rs

```diff
@@ -548,14 +551,76 @@ impl<'a> ContractRunner<'a> {
         let mut evm = InvariantExecutor::new(
             self.executor.clone(),
             runner,
-            invariant_config,
+            invariant_config.clone(),
             identified_contracts,
             known_contracts,
         );
-
         let invariant_contract =
             InvariantContract { address, invariant_function: func, abi: &self.contract.abi };
 
+        let mut logs = logs.clone();
+        let mut traces = traces.clone();
+        let mut coverage = coverage.clone();
+
+        // Try to replay recorded failure if any.
+        let failure_file = invariant_config
+            .clone()
+            .failure_file(self.name, invariant_contract.invariant_function.clone().name);
+        if failure_file.exists() {
+            if let Ok(data) = fs::read_to_string(failure_file) {
+                if let Ok(call_sequence) = serde_json::from_str::<Vec<BaseCounterExample>>(&data) {
```

Comment:
I believe there's a foundry common fs function for reading json, perhaps we can use this here and get rid of one scope?

Reply:
oh, yes, indeed, I used them for both read and write in https://github.com/foundry-rs/foundry/pull/7899/commits/9b220808c198d4831c5abed6c1c13b9e5e10b7c4